File: 2 Array:List/4.Linear_Search.py
def linear_search(arr, target):
    for i in range(len(arr)):
        if arr[i] == target:
            return i  # Return the index if element is found
    return -1  # Return -1 if element is not found

arr = [10, 20, 30, 40, 50]
target = 30
result = linear_search(arr, target)

if result != -1:
    print(f"Element {target} found at index {result}")
else:
    print("Element not found")


print("The output based on method: ", arr.index(target))

File: 2 Array:List/4.linearSearch.cpp
//Linear search
// arr[]={1,2,3,4} target=8
#include<iostream>
using namespace std;

int linearSearch(int arr[], int size, int target){
    for (int i=1; i<size; i++){
        if (arr[i]==target){
            return i;
        }
    }
    return -1;
}

int main() {
    int arr[] = {4,2,7,8,1,2,5};
    int target = 8;
    int size = sizeof(arr)/sizeof(arr[0]);
    int index = linearSearch(arr, size, target);

    cout<<"The target is in index: "<<index;
    return 0;
}